### PR TITLE
Add BucketTypeProperties Commands

### PR DESCRIPTION
**Pull request by alexmoore**

### CODE REVIEW

File: src/RiakClient/Models/RiakBucketProperties.cs [hunk 1]
```diff
@@ -744,42 +741,11 @@ private static IRiakPreCommitHook LoadPreCommitHook(RpbCommitHook hook)
                 hook.modfun.function.FromRiakString());
         }
 
-        private static IRiakPreCommitHook LoadPreCommitHook(JObject hook)
```
Comment (lukebakken) [inline src/RiakClient/Models/RiakBucketProperties.cs:747]: Dead code?

File: src/RiakClient/Models/RiakBucketProperties.cs [hunk 2]
```diff
@@ -191,27 +189,26 @@ internal RiakBucketProperties(RpbBucketProps bucketProps)
...
 
         /// <summary>
-        /// Indicates whether Legacy (Riak 1.0) Search is enabled for this bucket.
+        /// Indicates whether Legacy (Riak 1.0) Search is enabled for this bucket or bucket type.
         /// </summary>
         [Obsolete("Search is deprecated, please use LegacySearch instead.", true)]
         public bool? Search
         {
             get { return LegacySearch; }
-            private set { LegacySearch = value; }
```
Comment (lukebakken) [inline src/RiakClient/Models/RiakBucketProperties.cs:210]: Unused?

File: src/Test/Integration/KV/FetchPreflistTests.cs [hunk 3]
```diff
@@ -16,7 +16,7 @@
 // under the License.
 // </copyright>
 
-namespace Test.Integration.CRDT
```
Comment (lukebakken) [inline src/Test/Integration/KV/FetchPreflistTests.cs:19]: Good find

File: src/RiakClient/Models/RiakBucketProperties.cs (hunk 2 above)
Comment (alexmoore) [reply src/RiakClient/Models/RiakBucketProperties.cs:210]: Yeah, it's a `private set` and an alias to the new setter, and was also unused.

File: src/RiakClient/Models/RiakBucketProperties.cs (hunk 1 above)
Comment (alexmoore) [reply src/RiakClient/Models/RiakBucketProperties.cs:747]: Yep, we don't use the `JObject` overloads anymore, and it's private, so safe to delete.

File: src/Test/Integration/KV/FetchPreflistTests.cs (hunk 3 above)
Comment (alexmoore) [reply src/Test/Integration/KV/FetchPreflistTests.cs:19]: Yeah, just noticed that out of the blue.  I think there was a typo in one of the copyright headers too somewhere that I fixed.

**Review by lukebakken — APPROVED** [review]

**Pull request merged**